options.go

```diff
@@ -0,0 +1,66 @@
+package tflog
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type Option func(loggerOpts) loggerOpts
+
+type loggerOpts struct {
+	name            string
+	level           hclog.Level
+	includeLocation bool
+
+	// some private options to be used only by tflog for testing purposes
+	// we should never export an Option that sets these
+	output io.Writer
+}
+
+func applyLoggerOpts(opts ...Option) loggerOpts {
+	// set some defaults
+	l := loggerOpts{
+		level:           hclog.Trace,
+		includeLocation: true,
+	}
+	for _, opt := range opts {
+		l = opt(l)
+	}
+	return l
+}
+
+func withOutput(output io.Writer) Option {
+	return func(l loggerOpts) loggerOpts {
+		l.output = output
+		return l
+	}
+}
+
+func WithLogName(name string) Option {
```

Comment:
Nit: It would be great to see some (even terse) Go doc comments added for exported functions 👍

Reply:
Yeah, sorry, that was the plan before merging this, just didn't want to add a bunch of polish before verifying the direction we were heading. Added some docstrings.